imgix/luminous
# feat: add lib and es6 bundles, use closure compiler, remove gulp

## Description

This PR improves the build step of this library. The minified JS build is reduced from `14K -> 10K = 30% smaller`. The css size is unchanged.

There have been some changes to the source code to support the closure compiler. There might be a chance that things that worked before may not work now due to the optimisations it performs. I have tried to cover all common usage, but there may be some edge cases which I have not tested. All the tests still pass, and I've ensured that the tests test the compiled code.

Gulp has also been removed in favour of npm scripts.

### New Feature

- [x] If this is a big feature with breaking changes, consider [opening an issue][issues] to discuss first. This is completely up to you, but please keep in mind that your PR might not be accepted.

## Steps to Test

1. Clone this PR and run `npm install` to update dependencies
2. Run `npm run build` and open `index.html` in a browser, and test that Luminous still works normally
3. Create a consuming package, npm link, and ensure that Luminous works when importing using `import {Luminous} from 'luminous'`
